### ISSUE TITLE
FEATURE REQUEST: delete images from history view

### ISSUE THREAD

**Issue by dimateos** [ShareX/ShareX#6964]
You can delete files from the task list view in the main window:

![image](https://github.com/ShareX/ShareX/assets/26868611/6d05ae6f-284d-43f2-b5d9-50501381d12c)

But not from history views:

![image](https://github.com/ShareX/ShareX/assets/26868611/7188aba3-5235-4b90-ad67-fafa460aeafd)

❓ I wonder why not using the same menu with all options and icons (removing task related ones).

* You can easily filter images in the history views, then you can select some and operate on them. In contrast, you cannot filter the tasks in the main window.
* I feel like it would be very useful if at least the user could delete images from the filtered history view.
* Maybe individual OCR too, I rarely use it tho

**Comment by pixelinkmedia** [comment]
I would like to see this too.

**Comment by BirukTes** [comment]
Thank you for starting this, I have been awaiting something to done about this window/app for past 2 years or so. 

Image history is my most used feature (as well as Image Editor), it is where I organise my screenshots, but I learned to adjust with it, because it lacks key features, like deletion, and not syncing with folder history (images in screenshots folder vs Image history, it mostly occurs when Images are saved from Image Editor). Also, if an image is deleted from folder it shows as "missing" which can be filtered but it looks more clatter than any useful feature.

![image](https://github.com/ShareX/ShareX/assets/23151263/9efda8e1-e873-4d3e-904d-7e51ed875f6b)

**Comment by BirukTes** [comment]
Hi guys,

I have started on this feature, I have done minimum work for my use for now, but I will add more features in this area. The main issue was there was no way to update the history JSON file, only append, now can we fully update it which will allow for even more features to be made such as custom attributes and comments... Feel free to give feedback:
https://github.com/BirukTes/ShareX/tree/HistoryDeleteRemoveFileFeature

**Comment by X88R88** [comment]
> Hi guys,
> 
> I have started on this feature, I have done minimum work for my use for now, but I will add more features in this area. The main issue was there was no way to update the history JSON file, only append, now can we can fully update it, which will allow for even more features to be made such as comments, tags, and custom attributes... Feel free to give feedback: https://github.com/BirukTes/ShareX/tree/HistoryDeleteRemoveFileFeature

I just import the contents in this zip and replace the contents in the sharex folder?

**Comment by BirukTes** [comment]
> I just import the contents in this zip and replace the contents in the sharex folder?

No, there is no zip release for this yet, you will need to build it using the Release config first, then I recommend creating a shortcut to your desktop or start menu to the Release build exe output, then whenever you need to clean up your images, you run this modified app instead of the one you have installed... I am bit busy just now but I plan to complete these features and do a PR, it will be easier for everyone to have it as part of the main project.

**Comment by gdragon-git** [comment]
Did this feature ever make it  back to mainline trunk?

**Comment by BirukTes** [comment]
@gdragon-git No... from my side I have not yet made any further progress and so have not gone for PR.

**Comment by BirukTes** [comment]
I have create PR now: https://github.com/ShareX/ShareX/pull/7852
